### PR TITLE
[AutoMM] Fix NER HPO

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*

*Description of changes:*
`filter_hyperparameters ` requires column type `text_ner` to keep the `ner_text` model. Post-process the `column_types` from `infer_column_types` to replace column type `text` with `text_ner`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 1]
```diff
@@ -710,6 +711,9 @@ def fit(
             label_columns=self._label_column,
             provided_column_types=column_types,
         )
+        if self._problem_type == NER:
+            column_types = infer_ner_column_type(column_types=column_types)
```
Comment (liangfu) [inline multimodal/src/autogluon/multimodal/predictor.py:715]: Can we make this problem_type agnostic ?

**Review by liangfu — APPROVED** [review]
LGTM, thanks for the contribution. This unifies the HPO support for all problem types.

File: multimodal/src/autogluon/multimodal/predictor.py (hunk 1 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/predictor.py]: Good suggestion! Moved `infer_ner_column_type ` into `infer_column_types`.

**Review by cheungdaven — APPROVED** [review]
lgtm

**Pull request merged**